types/index.d.ts

```diff
@@ -944,4 +954,47 @@ declare namespace CbServer {
 
   type UserRolesRoleAddedTrigger = MakeUserRolesTrigger<"UserRoleAdded">;
   type UserRolesRoleRemovedTrigger = MakeUserRolesTrigger<"UserRoleRemoved">;
+
+  interface ProcessEnvOptions {
+    uid?: number | undefined;
+    gid?: number | undefined;
+    cwd?: string | URL | undefined;
+    env?: NodeJS.ProcessEnv | undefined;
+  }
+
+  type IOType = "overlapped" | "pipe" | "ignore" | "inherit";
+
+  type Stream = import("node:stream").Stream;
+  type StdioOptions =
+    | IOType
+    | Array<IOType | "ipc" | Stream | number | null | undefined>;
+  interface CommonOptions extends ProcessEnvOptions {
+    windowsHide?: boolean | undefined;
+    timeout?: number | undefined;
+  }
+
+  interface CommonExecOptions extends CommonOptions {
+    input?: string | NodeJS.ArrayBufferView | undefined;
+    stdio?: StdioOptions | undefined;
+    killSignal?: NodeJS.Signals | number | undefined;
+    maxBuffer?: number | undefined;
+    encoding?: BufferEncoding | "buffer" | null | undefined;
+  }
+  interface ExecSyncOptions extends CommonExecOptions {
+    shell?: string | undefined;
+  }
+  interface ExecSyncOptionsWithStringEncoding extends ExecSyncOptions {
+    encoding: BufferEncoding;
+  }
+  interface ExecSyncOptionsWithBufferEncoding extends ExecSyncOptions {
```

Comment:
Looks like this isn't used anywhere. Should it be used or did you want to keep it around in case we add support for it later?

Reply:
Hey Clark, by "not used anywhere", do you mean that we only use
`child_process.execSync(command)` and not `child_process.execSync(command, options)`, so that there is no need for `ExecSyncOptionsWithStringEncoding`? 

If that is the case, I do actually use the latter `execSync` options param. In fact, I think calling just `child_process.execSync(command)` without passing in at least an empty bracket for `options` throws an error in our code service (bug?).

Reply:
I was pointing out that `ExecSyncOptionsWithBufferEncoding` isn't used anywhere. Should it be used?